scripts/globals/beastmentreasure.lua

```diff
@@ -0,0 +1,394 @@
+        [771] = 1, -- Yellow rock
+        [772] = 1, -- Green rock
+        [773] = 1, -- Translucent rock
+        [774] = 1, -- Purple rock
+        [775] = 1, -- Black rock
+        [776] = 1, -- White rock
+        -- Tier 1 gems
+        [795] = 2, -- Lapis Lazuli (blue)
+        [796] = 2, -- Light Opal   (white)
+        [799] = 2, -- Onyx         (black)
+        [800] = 2, -- Amethyst     (purple)
+        [806] = 2, -- Tourmaline   (green)
+        [807] = 2, -- Sardonyx     (red)
+        [809] = 2, -- Clear Topaz  (translucent)
+        [814] = 2, -- Amber        (yellow)
+    },
+    seeds =
+    {
+        [572] = 1, -- Herb seeds
+        [573] = 1, -- Vegetable seeds
+        [574] = 1, -- Fruit seeds
+        [575] = 1, -- Grain seeds
+    },
+    coins =
+    {
+        [748] = 1, -- Gold beastcoin
+        [749] = 1, -- Mythril beastcoin
+        [750] = 1, -- Silver beastcoin
+    },
+    seals =
+    {
+        [1126] = 2 -- Beastmen's Seal
+    }
+}
+
+local function convertToWeighted(loot)
+    -- Takes a loot table formatted like the members of `sharedloot` and returns a weighted version that can be indexed
+    local weightedLoot = {}
+
+    for item, weight in pairs(loot) do
+        for i = 1, weight do
+            table.insert(weightedLoot, item)
+        end
+    end
+
+    return weightedLoot
+end
+
+local function addLoot(t1, t2)
+    -- Used for non-destructively combining a table containing a single weighted item (`t2`)
+    -- and an existing loot table containing one or more weighted items (`t1`).
+    newTable = {}
+    newItem = table.maxn(t2)
+
+    -- Copy the contents of the first table so that we don't modify the global version
+    for item, weight in pairs(t1) do
+        newTable[item] = weight
+    end
+    -- Add in our new item
+    newTable[newItem] = t2[newItem]
+
+    return newTable
+end
+
+-- Generate weighted loot tables now (on server init) so it doesn't have to be done at runtime
+local w_rocksgems   = convertToWeighted(sharedLoot.rocksgems)
+local w_seedsracial =
+{
+    [123] = convertToWeighted(addLoot(sharedLoot.seeds, zoneData[123].loot.racial)),
+    [124] = convertToWeighted(addLoot(sharedLoot.seeds, zoneData[124].loot.racial)),
+    [125] = convertToWeighted(addLoot(sharedLoot.seeds, zoneData[125].loot.racial))
+}
+local w_coins       = convertToWeighted(sharedLoot.coins)
+local w_sealsunique =
+{
+    [123] = convertToWeighted(addLoot(sharedLoot.seals, zoneData[123].loot.unique)),
+    [124] = convertToWeighted(addLoot(sharedLoot.seals, zoneData[124].loot.unique)),
+    [125] = convertToWeighted(addLoot(sharedLoot.seals, zoneData[125].loot.unique)),
+}
+
+local function weightedRandomSelect(w_loot)
+    -- Takes a weighted loot table and returns a single random result
+    return w_loot[math.random(#w_loot)]
+end
+
+local function getZoneID(entity)
+    return entity:getZone():getID()
+end
+
+local function getZoneData(entity)
+    return zoneData[getZoneID(entity)]
```

Comment:
I mean, while we're at it, this is superfluous too.

```lua
zoneData[player:getZoneID()
```

You don't need to abstract functionality built in to lua for accessing data.

Reply:
Removed this along with its ugly sister function in https://github.com/DarkstarProject/darkstar/pull/6095/commits/9ab5061d14b3673af6a49ec81d38dcc1f1012eb8.